services/swagger.js

```diff
@@ -21,10 +21,10 @@ var httpMethods = ['get', 'put', 'post', 'delete', 'options', 'head', 'patch'];
 
 var responseModelValidationLevel;
 var polymorphicValidation;
+var refCompilation;
 
-exports.init = function (logger, config, callback) {
+exports.init = function (logger, config, refCompiler, callback) {
```

Comment:
the config could be passed to the ref compiler function ... similarly the logger for any logging it needs to do